code/game/objects/items/grenades/flashbang.dm

```diff
@@ -21,16 +21,19 @@
 /obj/item/grenade/flashbang/proc/bang(turf/T , mob/living/M)
 	if(M.stat == DEAD)	//They're dead!
 		return
-	M.show_message("<span class='warning'>BANG</span>", 2)
+	M.show_message("<span class='userdanger'>BANG</span>", 2)
 	var/distance = max(0,get_dist(get_turf(src),T))
-
-//Flash
-	if(M.flash_act(affect_silicon = 1))
-		M.Paralyze(max(200/max(1,distance), 60))
-//Bang
 	if(!distance || loc == M || loc == M.loc)	//Stop allahu akbarring rooms with this.
 		M.Paralyze(200)
-		M.soundbang_act(1, 200, 10, 15)
+		M.soundbang_act(1, 20, 10, 15)
```

Comment:
This means if you're on top of the flashbang when it goes off, you will be stunned for far less.

This lets you be a flashbang suicide bomber again, basically.

Reply:
I actually changed the definition of soundbang_act.  Paralyze() works in deciseconds, but confusion works in seconds.  You can see in the code that I changed every other use of the function too.

Flashbang suicide bombing is degenerate but 200 seconds of confusion is a step too far.  Of course, since it completely overlays the stun, it'll only affect someone with stun reduction.